ds1307/ds1307.go

```diff
@@ -0,0 +1,145 @@
+package ds1307
+
+import (
+	"time"
+
+	"machine"
+)
+
+// Device wraps an I2C connection to a DS1307 device.
+type Device struct {
+	bus        machine.I2C
+	address    uint8
+	addresSRAM uint8
```

Comment:
This should also probably be public. Also, should probably be spelled `AddressSRAM` as weird as that looks.